bazel/envoy_internal.bzl

```diff
@@ -154,6 +154,13 @@ def envoy_stdlib_deps():
         "//conditions:default": ["@envoy//bazel:static_stdlib"],
     })
 
+def envoy_dbg_linkopts():
+    return select({
+        # TODO: Remove once we have https://github.com/bazelbuild/bazel/pull/15635
+        "@envoy//bazel:apple_non_opt": ["-Wl,-no_deduplicate"],
```

Comment:
naming is hard, dbg, and non_opt, i don't like either, up for suggestions here. Really I want a single word that conveys both "fastbuild and debug"